File: Go/checkout_get_order.go
package main

import (
	"crypto/sha512"
	"encoding/base64"
	"fmt"
	"net/http"
	"time"
	"io/ioutil"
)

// Adjust SveaBasicSettings based on your credentials.
type TestStruct struct {
	SveaBasicSettings struct {
		MerchantID string
		SecretKey  string
	}
}

func main() {
	testInstance := TestStruct{}

	url := "https://checkoutapistage.svea.com/api/orders/1"

	mySCOHeaders := testInstance.getRequestHeaders(map[string]string{"Connection": "keep-alive"})
	// fmt.Println(mySCOHeaders)

	// Send request
	payload := []byte{}
	response, err := http.Get(url)
	if err != nil {
		fmt.Println("Error sending request:", err)
		return
	}
	defer response.Body.Close()

	body, err := ioutil.ReadAll(response.Body)
	if err != nil {
		fmt.Println("Error reading response body:", err)
		return
	}
	fmt.Println(string(body))

	if response.StatusCode != http.StatusOK {
		fmt.Println("Request error:", response.Status)
		return
	}
}

func (t TestStruct) getRequestHeaders(extraHeaders map[string]string) map[string]string {
	timestamp := time.Now().Format("2006-01-02 15:04:05")
	token := t.getAuthToken(timestamp, "")
	headers := map[string]string{
		"Authorization": "Svea " + token,
		"Timestamp":     timestamp,
	}

	for key, value := range extraHeaders {
		headers[key] = value
	}

	return headers
}

func (t TestStruct) getAuthToken(timestamp, requestBody string) string {
	digest := requestBody + t.SveaBasicSettings.SecretKey + timestamp
	h := sha512.New()
	h.Write([]byte(digest))
	sha := h.Sum(nil)
	merchantAuth := t.SveaBasicSettings.MerchantID + ":" + fmt.Sprintf("%x", sha)
	token := base64.StdEncoding.EncodeToString([]byte(merchantAuth))
	return token
}

File: Python/checkout_get_order.py
from datetime import datetime
from datetime import timezone
from pandas import json_normalize
import hashlib
import base64
import requests

order_id = "8906830"
merchant_id = "124842"
secret_word = "1NDxpT2WQ4PW6Ud95rLWKD98xVr45Q8O9Vd52nomC7U9B18jp7lHCu7nsiTJO1NWXjSx26vE41jJ4rul7FUP1cGKXm4wakxt3iF7k63ayleb1xX9Di2wW46t9felsSPW"

class TestClass:

    def __init__(self):
        pass

    def get_request_headers(self, requests_body="", extra_headers=None):
        #timestamp: str = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        timestamp: str = datetime.now(timezone.utc).strftime("%Y-%m-%d %H:%M:%S")
        token = self.get_auth_token(timestamp, requests_body)
        #print(token)
        headers = {"Authorization": f"Svea {token}", "Timestamp": timestamp}
        if extra_headers:
            headers.update(extra_headers)
        return headers

    def get_auth_token(self, timestamp, request_body="") -> str:
        digest = request_body + secret_word + timestamp
        h = hashlib.sha512(digest.encode("utf-8")).hexdigest()
        auth = f"{merchant_id}:{h}"
        token = base64.b64encode(f"{auth}".encode('ascii')).decode('ascii')
        return token

testInstance = TestClass()
#my_headers = testInstance.get_request_headers(extra_headers={'Content-Type':'application/json'})
my_headers = testInstance.get_request_headers()
print(my_headers)

url = "https://checkoutapistage.svea.com/api/orders/" + order_id
payload = {}

print("Fetching order:", order_id)
response = requests.request("GET", url, headers=my_headers, data=payload)
#response = requests.get(url, headers=my_headers, data=payload)
print(response.text)
#response.raise_for_status()

if response:
    dictr = response.json()
    result = json_normalize(dictr)
    print(result)
